src/panels/lovelace/editor/hui-dialog-edit-card.ts

```diff
@@ -59,42 +67,104 @@ export class HuiDialogEditCard extends LitElement {
-            .yaml="${this._cardConfig}"
-            @yaml-changed="${this._handleYamlChanged}"
-          ></hui-yaml-editor>
+          ${
+            // Look to see if this has already be done if yes dont use until just render this._elementConfig
+            this._showUIEditor && !this._elementConfig
+              ? html`<div class="element-editor">${until(
+                  this._loadConfig().then(() => this._loadElementConfig()),
+                  html`<span>Loading...</span>`
+                )}</div>`
+              : this._showUIEditor
+                ? html`<div class="element-editor">${this._elementConfig}</div>`
+                : html`
+                  <hui-yaml-editor
+                    .yaml="${this._cardConfig}"
+                    @yaml-changed="${this._handleYamlChanged}"
+                  ></hui-yaml-editor>`
+          }
           <hui-yaml-card-preview
             .hass="${this.hass}"
             .yaml="${this._cardConfig}"
           ></hui-yaml-card-preview>
         </paper-dialog-scrollable>
         <div class="paper-dialog-buttons">
-          <paper-button @click="${this._closeDialog}">Cancel</paper-button>
-          <paper-button @click="${this._updateConfig}">Save</paper-button>
+          <paper-button
+            @click="${this._toggleEditor}"
+          >Toggle Editor</paper-button>
+          <paper-button
+            @click="${this._closeDialog}"
+          >Cancel</paper-button>
+          <paper-button
+            @click="${this._updateConfig}"'
+          >Save</paper-button>
         </div>
       </paper-dialog>
     `;
   }
 
   private _handleYamlChanged(ev) {
-    this._previewEl.yaml = ev.detail.yaml;
+    this._handleConfigChanged("yaml", ev.detail.yaml);
+  }
+
+  private _handleConfigChanged(type: string, changedProps: any): void {
+    if (!this._previewEl) {
+      return;
+    }
+
+    this._previewEl.value = { format: type, value: changedProps };
   }
 
   private _closeDialog() {
     this._dialog.close();
   }
 
+  private _toggleEditor() {
+    if (!this._elementConfig) {
+      return;
+    }
+    this._showUIEditor = !this._showUIEditor;
+    // This will center the dialog with the updated config
+    fireEvent(this._dialog, "iron-resize");
+  }
+
   private async _loadConfig() {
     this._cardConfig = await getCardConfig(this.hass!, this._cardId!);
     await this.updateComplete;
     // This will center the dialog with the updated config
     fireEvent(this._dialog, "iron-resize");
   }
 
+  private async _loadElementConfig(): Promise<LovelaceCardEditor> {
+    const conf = yaml.safeLoad(this._cardConfig);
+    const elClass = customElements.get(`hui-${conf.type}-card`);
+    let elementConfig;
+    try {
+      elementConfig = await elClass.getConfigElement();
+    } catch (err) {
+      // No Element Config Function on Element
+      this._showUIEditor = false;
+      return undefined;
+    }
+    if (!elementConfig) {
+      this._showUIEditor = false;
```

Comment:
Don't change this value. Instead, the render method should be able to derive from the state of the properties if it should show a "loading text" (`this._configElement == undefined`) or render the YAML editor (`this._configElement == null`)

`undefined` -> we haven't tried loading one
`null` -> we tried loading one, it failed or there is no method or it doesn't exist

Reply:
Good call

Reply:
But this doesn't work with a toggle unless I Null the value of _elementConfig right?

Reply:
you still need this boolean around, just don't change it if there is no config.